kapajira/kapacitor/utils.py

```diff
@@ -0,0 +1,34 @@
+import json
+from collections import abc
+
+
+class AlertDataParser:
+    @classmethod
+    def parse(cls, alert_data):
+        alert_dict = json.loads(alert_data)
+        return AlertData(alert_dict)
+
+
+class AlertData:
+    """ A read-only façade for navigating a AlertNode JSON object
+        using attribute notation
+    """
+
+    def __init__(self, mapping):
+        self._data = dict(mapping)
+
+    def __getattr__(self, name):
+        if hasattr(self._data, name):
```

Comment:
Is this result cached between calls to the getter? I mean if one caller fetches attribute 'foo' and a Mapping or a list is returned. And (for any reason) the caller modifies its copy - does it alter the result of the next call to a 'foo' getter? And a follow-up question - do we/should we have a test for it :)